packages/react/src/Delay.tsx

```diff
@@ -1,17 +1,21 @@
-import { type PropsWithChildren, useContext, useState } from 'react'
+import { type PropsWithChildren, type ReactNode, useContext, useState } from 'react'
 import { DelayDefaultPropsContext } from './contexts'
 import { useTimeout } from './hooks'
 
 export interface DelayProps extends PropsWithChildren {
   ms?: number
+  /**
+   * @experimental This is experimental feature.
+   */
+  fallback?: ReactNode
 }
 
-export const Delay = ({ ms, children }: DelayProps) => {
+export const Delay = ({ ms, children, fallback }: DelayProps) => {
   const defaultProps = useContext(DelayDefaultPropsContext)
   const delayMs = ms ?? defaultProps.ms ?? 0
   const [isDelayed, setIsDelayed] = useState(delayMs === 0)
   useTimeout(() => setIsDelayed(true), delayMs)
-  return <>{isDelayed ? children : null}</>
+  return <>{isDelayed ? children : typeof fallback === 'undefined' ? defaultProps.fallback : fallback}</>
```

Comment:
Rather than using nested trinomials, why not just define and use the fallback once above, or even use an early return strategy to simplify the return statement!